### PR TITLE
Add DeferredText.resolveToString extension

### LINKED ISSUE #20: `DeferredText` - support for `String` resolution
`DeferredText` is always resolved to `CharSequence`. It would be nice to add support to resolve to `String` as well.

### PR DESCRIPTION
**Pull request by drewhamilton**
Closes #20.

### CODE REVIEW

File: deferred-resources/src/androidTest/java/com/backbase/deferredresources/text/DeferredTextUtilsTest.kt [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package com.backbase.deferredresources.text
+
+import com.backbase.deferredresources.DeferredText
+import com.backbase.deferredresources.context
+import com.google.common.truth.Truth.assertThat
+import org.junit.Test
+
+class DeferredTextUtilsTest {
+
+    @Test fun resolveAsString_returnsCharSequenceToString() {
```
Comment (Hariofspades) [inline deferred-resources/src/androidTest/java/com/backbase/deferredresources/text/DeferredTextUtilsTest.kt:10]: small but... 
`resolveToString_returnsCharSequenceToString()`

File: deferred-resources/src/main/java/com/backbase/deferredresources/DeferredText.kt [hunk 2]
```diff
@@ -14,6 +14,11 @@ interface DeferredText {
      */
     fun resolve(context: Context): CharSequence
 
+    /**
+     * Resolve the text as a string.
+     */
+    fun resolveAsString(context: Context) = resolve(context).toString()
```
Comment (Hariofspades) [inline deferred-resources/src/main/java/com/backbase/deferredresources/DeferredText.kt]: I did not understand why we are keeping both interface and extension?

Comment (drewhamilton) [reply deferred-resources/src/main/java/com/backbase/deferredresources/DeferredText.kt]: We're not, you must have looked at an old commit

**Comment by Hariofspades** [comment]
Looks good

**Pull request merged**